sysstat.service.in

```diff
@@ -2,20 +2,27 @@
 # (C) 2012 Peter Schiffer (pschiffe <at> redhat.com)
 #
 # @PACKAGE_NAME@-@PACKAGE_VERSION@ systemd unit file:
-#	 Insert a dummy record in current daily data file.
-#	 This indicates that the counters have restarted from 0.
+#     1. Insert a dummy record in current daily data file when starting/stopping
+#        the unit. This indicates that the counters have restarted from 0.
+#     2. This unit is automatically stopped/started on suspend/resume,
+#        see /usr/lib/systemd/system-sleep/sysstat-sleep.sh.
+#     3. Starting this unit also starts the timer unit for periodic accounting.
 
 [Unit]
-Description=Resets System Activity Logs
+Description=System Activity Accounting
+# Start the periodic accounting process
+Requires=sysstat-collect.timer
+# but make sure we finish first, because the accounting process locks the datafile
+Before=sysstat-collect.timer sysstat-collect.service
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
 User=@CRON_OWNER@
-ExecStart=@SA_LIB_DIR@/sa1 --boot
+ExecStart=@SA_LIB_DIR@/sa1 --boot -C STARTED
```

Comment:
The LINUX-RESTART message must be inserted in a sar binary datafile whenever the system is restarted. Indeed, many Linux kernel counters are absolute values since system startup. For example, the counters used to calculate CPU utilization (%user, %nice, etc.) contain the absolute number of jiffies spent in the corresponding mode (user, nice...) since system startup. So imagine the following sequence (T1<T2<T3):
T1 : #user_jiffies=1000
T2: system reboot
T3: #user_jiffies=1500
Without the LINUX-RESTART message, sar will calculate %user at T3 thinking that the CPU has spent **500** jiffies in user mode between **T1** and T3. Which is wrong.
With the LINUX-RESTART message, sar will know that the system has rebooted and that the CPU has spent **1500** jiffies in user mode between **T2** and T3 and calculate %user at T3 accordingly. The result will be correct (and quite different from the previous case).
Note that LINUX-RESTART message is not saved in file as a comment: This is really a special record indicating that a system restart has occurred, and displayed by sar as LINUX-RESTART when displaying the file contents.

Wrt suspend/resume operations, I don't know whether counters restart from zero when you resume. Personally I would say no. In this case, inserting a comment like STARTED or STOPPED should be just fine.